tools/targets/PSOC6.py

```diff
@@ -0,0 +1,167 @@
+#
+# Copyright (c) 2017-2018 Future Electronics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import platform
+import subprocess
+import errno
+from array import array
+from distutils.spawn import find_executable
+from shutil import copyfile
+from intelhex import IntelHex
+from intelhex.compat import asbytes
+
+from ..config import ConfigException
+
+
+class HookError(Exception):
+    def __init__(self, msg):
+        self.msg = msg
+    def __str__(self):
+        return self.msg
+
+
+# Patch Cypress hex file:
+# - update checksum
+# - update metadata
+# - align regions to page (256 bytes) boundary
+def patch(message_func, ihex, hexf, align=256):
+    #calculate checksum
+    checksum = 0
+    for start, end in ihex.segments():
+        if start  >= 0x090000000:
+            continue
+        segment = ihex.tobinarray(start = start, end = end)
```

Comment:
Generally and elsewhere in this file, we don't put ` ` around `=` in the argument positions.